includes/admin/class-wc-rest-payments-webhook-controller.php

```diff
@@ -301,9 +301,18 @@ private function process_webhook_payment_intent_succeeded( $event_body ) {
 		// Get an updated set of order properties to avoid race conditions when the server sends the paid webhook before we've finished processing the original payment request.
 		$order->get_data_store()->read( $order );
```

Comment:
I'm wondering if we can get rid of this now or do you think it's still valuable? Originally I added this right before the status check so we get the most recent up to date status incase it was processed via a different request already. If we're going to be locking the order via a transient, then that might not be needed anymore.

Reply:
I just realised this question might tie into your reasoning behind keeping the status check below.

Reply:
> I just realised this question might tie into your reasoning behind keeping the status check below.

That's correct, I don't have a strong opinion against removing the status check and associated code, opted to keep it as an additional check that bails out a little early. I also don't know whether the extra read has a measurable impact on performance. Happy to get rid of it if you think it has a negative impact :)

Reply:
yeah that's all good. 

I don't think it has a measurable impact. There are a lot of order reads that happen in each request so this would be very minor in the grand scheme of things. I also cannot simulate potential scenarios where this would be good, bad or indifferent. Happy to err on the side of caution and keep it. :)